In [1]:
import pandas as pd

In [2]:
url = "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/titanic.csv"
df = pd.read_csv(url)
df

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

In [3]:
df.shape

(891, 15)

2.2 Observations: refers to a single row of the dataset (example: details about specific character)

Variables: refers to columns of the dataset, containing specific type of information about all of the observations (example: attributes of the characters)

In [4]:
df.describe()

survived      pclass         age       sibsp       parch        fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

In [5]:
df['sex'].value_counts()

sex
male      577
female    314
Name: count, dtype: int64

4a. There are only 6 columns shown using df.describe() code, while the df.shape shows that there are actually 15 columns in the dataset.

4b. The df.shape code provides 891 as the total rows in the dataset. However, the df.describe() shows that there are only 714 values in the count columns, under the age category.

5. Method (as in df.shape) directly gives information stored in the dataframe without doing any calculation, while attributes (as in df.describe()) performs a set of operations on the data stored in the dataframe, which is the reason why we need the parentheses in the end (because it calculates the data)

6. count = the amount of not-empty values

mean = the average values

std = the standard deviations

min = the smallest value

25% = first quartile

50% = second quartile; median

75% = third quartile

max = the biggest value

7.1 The case is that I want to analyze the data from the Titanic dataset, but some rows have some missing values, for example in columns "age". I think that the other values are important, thus only the missing value is needed to be removed, then df.dropna() is preferred.

7.2 In the columns "deck", most of the values are missing, so there is no point of keeping it; we can't get a proper analysis from those data. In this case, the del df['col'] is preferred.

7.3 It is necessary to use del df['col'] first before df.dropna() because we can avoid removing important data for our analysis. Let's say that we use df.dropna() first, it will remove every rows that has missing value; in the column "deck", many rows will be removed since there are many missing values there, making us to remove important data from other columns as well. However, if we use del df['col'] to remove column "deck" first, then applying df.dropna(), the only rows that will be deleted are those that are not too important, which has missing values in it.

In [6]:
df.isna().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

In [7]:
del df['deck']
df = df.dropna()
df.isna().sum()

survived       0
pclass         0
sex            0
age            0
sibsp          0
parch          0
fare           0
embarked       0
class          0
who            0
adult_male     0
embark_town    0
alive          0
alone          0
dtype: int64

7.4 Dropping the 'deck' column first prevent the unnecessary removal of rows since it has so many missing values (688 values). Then, we can continue to remove any missing values using df.dropna() to clear our dataset. Before, there are missing values under the columns 'age', 'embarked', 'embark_town', and 'deck'. After, the missing values under the 'age', 'embarked', 'embark_town', and the columns 'deck' were dropped.

8.1 The df.groupby("col1")["col2"].describe() breakdowns the distribution of 'col2' within each unique value of 'col1'

In [9]:
df.groupby('pclass')['survived'].describe()

count      mean       std  min  25%  50%  75%  max
pclass                                                    
1       184.0  0.652174  0.477580  0.0  0.0  1.0  1.0  1.0
2       173.0  0.479769  0.501041  0.0  0.0  0.0  1.0  1.0
3       355.0  0.239437  0.427342  0.0  0.0  0.0  0.0  1.0

8.2 Before cleaning, the missing values are still included in the calculations performed, thus affecting the final results. However, after cleaning the missing values, it reflects the actual calculations which is more accurate.

8.3 It is quicker to use ChatBot since it directly gives the solutions, and mostly, they are accurate. Meanwhile, if using Google, we need to open websites and read the explanation, which is more time-consuming.

9. Yes